### PR TITLE
Add a proper warning if the add-on exceeds configured maxSize.

### LINKED ISSUE #5752: Wrong message displayed when trying to upload an add-on with over 200 MB
Steps to reproduce:
1.Go to AMO Stage and try to upload an add-on bigger than 200MB(https://www.dropbox.com/s/03u7sgefn8r7wsj/testbigsize2.zip?dl=0)

Expected results:
Validation is failed and the error message should be useful

Actual results:
Validation is failed , but the validation message is not useful for understanding the reasons : "Received an empty response from the server; status: 0"

Please see screenshot for this issue : 
over200mb

### PR DESCRIPTION
**Pull request by EnTeQuAk**
* adds a generic maxSize setting to our file uploader
* catches xhr 413 responses in case our javascript didn't catch
file-size and nginx aborts

Fixes #5752

Our JavaScript generically catching the file size:
screenshot from 2017-07-06 12-26-35

Nginx catching it, responds with HTTP 413 and we're catching it:
screenshot from 2017-07-06 10-43-33

### CODE REVIEW

File: static/js/common/upload-addon.js [hunk 1]
```diff
@@ -27,7 +27,12 @@
     }
 
     $.fn.addonUploader = function( options ) {
-        var settings = {'filetypes': ['zip', 'xpi', 'crx', 'jar', 'xml'], 'getErrors': getErrors, 'cancel': $()};
+        var settings = {
+            'filetypes': ['zip', 'xpi', 'crx', 'jar', 'xml'],
+            'getErrors': getErrors,
+            'cancel': $(),
```
Comment (eviljeff) [inline static/js/common/upload-addon.js]: I know this is existing code you're just reformatting, but having cancel as `$()` seems super weird to me.

File: static/js/common/upload-addon.js [hunk 2]
```diff
@@ -277,9 +282,18 @@
                     $upload_field.trigger("upload_finished", [file]);
 
                 } else if(xhr.readyState == 4 && !aborted) {
-                    // L10n: first argument is an HTTP status code
-                    errors = [format(gettext("Received an empty response from the server; status: {0}"),
-                                     [xhr.status])];
+                    if (xhr.status == 413) {
+                        errors.push(
+                            format(
+                                gettext("Your add-on exceeds the maximum size of 200 MB."),
```
Comment (eviljeff) [inline static/js/common/upload-addon.js]: use the `textSize()` function for the size, like below.

File: static/js/common/upload-base.js [hunk 3]
```diff
@@ -6,7 +6,15 @@
         return results.errors;
     }
 
-    var settings = {'filetypes': [], 'getErrors': getErrors, 'cancel': $()};
+    function textSize(bytes) {
+        // Based on code by Cary Dunn (http://bit.ly/d8qbWc).
```
Comment (eviljeff) [inline static/js/common/upload-base.js]: Not that it matters, but that url is a 404.  Also, I despair that JS doesn't have a built-in method for this.

File: static/js/common/upload-addon.js (hunk 1 above)
Comment (EnTeQuAk) [inline static/js/common/upload-addon.js]: Well, that's the lazy variant of someone requiring a jquery-element (e.g `$('a.cancel')`) who don't want to bother checking if the option is given or not. `upload-addon.js` also re-implements some of the cancelation logic but right now I didn't want to fix this, I'd rather have a follow-up that refactors that code with proper tests…

File: static/js/common/upload-addon.js (hunk 2 above)
Comment (EnTeQuAk) [inline static/js/common/upload-addon.js]: Fixed in f2bb2aef31

File: static/js/common/upload-base.js (hunk 3 above)
Comment (EnTeQuAk) [inline static/js/common/upload-base.js]: python doesn't have one either, but yeah, I'll remove that hint.

Comment (EnTeQuAk) [inline static/js/common/upload-base.js]: Removed in 379cf1a456

**Pull request merged**